resotoworker/resotoworker/__main__.py

```diff
@@ -233,20 +234,27 @@ def core_actions_processor(
     data = message.get("data") or {}
     task_id: str = data.get("task")  # type: ignore
     step_name: str = data.get("step")  # type: ignore
+    timing: Json = data.get("metadata", {}).get("timing", {})
+    last_run_raw: Optional[int] = timing.get(step_name, {}).get("started_at", None)
+    last_run: Optional[datetime] = None
+    if last_run_raw:
+        last_run = datetime.fromtimestamp(last_run_raw, tz=timezone.utc)
+    metadata = CollectorMetadata(task_id, step_name, last_run)
```

Comment:
Why do we now have a CollectorMetadata()? There is already `data` in the `message`. Just pass the `data` dict as `task_data` to everywhere. Anybody interested in any of the task data can then choose the parts that interest them.